Build failure: Kstars

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
config = {
    environment = {
      systemPackages = with pkgs; [
        kstars
      ];
    };
  };
```

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

https://hydra.nixos.org/build/309620533

### Relevant log output

```console
error: Cannot build '/nix/store/6nyvjadyrlqrj53dbjhshlbi7dn65wi5-limesuite-23.11.0.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0
       Last 19 log lines:
       > Using udevCheckHook
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/bj3j9hs7kgaqifr23gzym6s8vcq2m9h5-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/share/doc/LimeSuite -DCMAKE_INSTALL_INFODIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/prqnqy7kp72vlm7wh1pxx3b8z59csq4m-limesuite-23.11.0 -DOpenGL_GL_PREFERENCE=GLVND -DENABLE_GUI=OFF
       > CMake Error at CMakeLists.txt:4 (cmake_minimum_required):
       >   Compatibility with CMake < 3.5 has been removed from CMake.
       >
       >   Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
       >   to tell CMake that the project requires at least <min> but has been updated
       >   to work with policies introduced by <max> or earlier.
       >
       >   Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
       >
       >
       > -- Configuring incomplete, errors occurred!
       For full logs, run:
         nix log /nix/store/6nyvjadyrlqrj53dbjhshlbi7dn65wi5-limesuite-23.11.0.drv
error: Cannot build '/nix/store/4rzq2wxmqkrz48nli179p5r2mv2fclv7-indi-3rdparty-indi-limesdr-2.1.5.1.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/6d34gnzkdbpngzmw8i0shgfqz1mqxcni-indi-3rdparty-indi-limesdr-2.1.5.1
error: Cannot build '/nix/store/ax8nxzhfc7lzzyr1sig6lc2a2fh45kwy-indi-full-2.1.5.1.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/w1pshgqn5i5m0gjhg3jn38r8mdayx9ls-indi-full-2.1.5.1
error: Cannot build '/nix/store/dvgwvlppqzimzw6k99s5hazm4v64rrdf-kstars-3.7.8.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/pl91ip28ip3gmdz8cvin8l88c55mbf6z-kstars-3.7.8
error: Cannot build '/nix/store/finj223cksjrvaksrfz2v7ls7cnypap0-system-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/i0im2zgqxzy8i9g1d06js1wi7cpvs3jq-system-path
error: Cannot build '/nix/store/vx3nbnz3nrx4pz5vl8zzs445hnkh6m2x-nixos-system-tycho-station-25.11.20251007.c9b6fb7.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/z74ar00g26abyj1i58ppa5r3gylgg0mg-nixos-system-tycho-station-25.11.20251007.c9b6fb7
```

### Additional context

The problem seems to be related to KStars dependencies indi and limesuite, tagging their maintainers as well, and it's related to the CMake changes of the past few days, so there might be relevant informations on recent merges solving other similar issues.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.16.9, NixOS, 25.11 (Xantusia), 25.11.20251002.7df7ff7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `""`
 - nixpkgs: `/nix/store/9v6qa656sq3xc58vkxslqy646p0ajj61-source`

### Notify maintainers

@markuskowa 
@sheepforce 
@returntoreality 
@TimPut 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/science/astronomy/indilib/default.nix
+++ b/pkgs/development/libraries/science/astronomy/indilib/default.nix
@@ -2,6 +2,7 @@
   stdenv,
   lib,
   fetchFromGitHub,
+  fetchpatch2,
   bash,
   cmake,
   cfitsio,
@@ -17,22 +18,36 @@
   fftw,
   gtest,
   udevCheckHook,
+  versionCheckHook,
   indi-full,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "indilib";
-  version = "2.1.5.1";
+  version = "2.1.6";
 
   src = fetchFromGitHub {
     owner = "indilib";
     repo = "indi";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-mbY3iDLRcQ+pis26u6pHzB43ureaKH7KYPkV0CwHU/E=";
+    hash = "sha256-0+ZC9NoanBDojYz/ufZUpUQB++vnMcUYtG1UmmVGbTg=";
   };
+
+  # fixes version number. This commit is directly after the tagged commit in master
+  # should be removed with the next release
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/indilib/indi/commit/91e3e35250126887a856e90b6a0a30697fb01545.patch?full_index=1";
+      hash = "sha256-ho1S+A6gTQ9ELy/QE14S6daXyMN+vASFbXa2vMWdqR8=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
     udevCheckHook
   ];
 
@@ -86,6 +101,7 @@ stdenv.mkDerivation (finalAttrs: {
     description = "Implementation of the INDI protocol for POSIX operating systems";
     changelog = "https://github.com/indilib/indi/releases/tag/v${finalAttrs.version}";
     license = licenses.lgpl2Plus;
+    mainProgram = "indiserver";
     maintainers = with maintainers; [
       sheepforce
       returntoreality

--- a/pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix
+++ b/pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix
@@ -9,7 +9,6 @@
   coreutils,
   cfitsio,
   fetchFromGitHub,
-  fetchpatch,
   gtest,
   libusb1,
   libusb-compat-0_1,
@@ -42,12 +41,13 @@
 }:
 
 let
+  thirdparty_version = "2.1.6.2";
   fxload = libusb1.override { withExamples = true; };
   src-3rdparty = fetchFromGitHub {
     owner = "indilib";
     repo = "indi-3rdparty";
-    rev = "v${indilib.version}";
-    hash = "sha256-+WBQdu1iWleHf6xC4SK69y505wqZ36IUM4xnh1fnc6s=";
+    rev = "v${thirdparty_version}";
+    hash = "sha256-FMvdm7dkOkRlmbPNeQjh0jd+2bOinzW13QPP2NnOr/M=";
   };
 
   buildIndi3rdParty =
@@ -58,7 +58,7 @@ let
       cmakeFlags ? [ ],
       postInstall ? "",
       doCheck ? true,
-      version ? indilib.version,
+      version ? thirdparty_version,
       src ? src-3rdparty,
       meta ? { },
       ...
@@ -551,14 +551,6 @@ in
   indi-astarbox = buildIndi3rdParty {
     pname = "indi-astarbox";
     buildInputs = [ indilib ];
-    # TODO patch already upstream, remove with version > 2.1.5.1
-    patches = [
-      (fetchpatch {
-        url = "https://github.com/indilib/indi-3rdparty/commit/c347000ec227a5ef98911aab34c7b08a91509cba.patch";
-        hash = "sha256-M3b4ySoGJRpfNmBaagjDaeEPKqwaVgRUWaQY626SGBI=";
-        stripLen = 1;
-      })
-    ];
   };
 
   indi-astroasis = buildIndi3rdParty {
@@ -713,15 +705,6 @@ in
       glib
       zlib
     ];
-    # TODO patch already upstream, remove with version > 2.1.5.1
-    patches = [
-      (fetchpatch {
-        url = "https://github.com/indilib/indi-3rdparty/commit/c33c08b50093698e2aa73d73783d96f85df488a9.patch";
-        hash = "sha256-EQ2G9gTexf9FESCAR28f2cwzvH4TOAA8bvyJCxFv/E8=";
-        stripLen = 1;
-      })
-    ];
-
   };
 
   indi-gphoto = buildIndi3rdParty {

